core-web/src/main/java/org/silverpeas/core/web/attachment/tag/SimpleDocumentContextualMenu.java

```diff
@@ -220,29 +197,84 @@ String prepareActions(SimpleDocument attachment, boolean useXMLForm, boolean use
       builder.append(configureCheckoutAndEdit(attachmentId, !useWebDAV || !attachment.
           isOpenOfficeCompatible()));
     }
-    builder.append(configureFileSharing(attachmentId,
-        !attachment.isSharingAllowedForRolesFrom(UserDetail.getById(userId))));
+    builder.append(configureFileSharing(attachmentId, !attachment.isSharingAllowedForRolesFrom(user)));
     builder.append(configureSwitchState(attachmentId, (!attachment.isVersioned() &&
         isComponentPublicationAlwaysVisible(attachment.getInstanceId())) ||
         attachment.isReadOnly()));
     builder.append(configureNotify(attachmentId, !showMenuNotif));
-    builder.append("YAHOO.util.Event.addListener(\"basicmenu").append(attachmentId);
-    builder.append("\", \"mouseover\", oMenu").append(attachmentId).append(".show);");
-    builder.append("YAHOO.util.Event.addListener(\"basicmenu").append(attachmentId);
-    builder.append("\", \"mouseout\", oMenu").append(attachmentId).append(".hide);");
-
-    if (!useContextualMenu) {
-      builder.append("YAHOO.util.Event.on(\"edit_").append(attachmentId);
-      builder.append("\", \"click\", function (event) {");
-      builder.append("var xy = YAHOO.util.Event.getXY(event);");
-      builder.append(oMenuId).append(".cfg.setProperty(\"x\", xy[0]);");
-      builder.append(oMenuId).append(".cfg.setProperty(\"y\", xy[1]+10);");
-      builder.append(oMenuId).append(".show();");
-      builder.append("  })");
+
+    String itemsConfig = builder.toString();
+
+    return getMenu(attachmentId, menuItems, itemsConfig);
+  }
+
+  String prepareReadOnlyActions(SimpleDocument attachment, UserDetail user,
+      LocalizationBundle resources, boolean showMenuNotif) throws UnsupportedEncodingException {
+    String attachmentId = String.valueOf(attachment.getOldSilverpeasId());
+    StringBuilder itemsBuilder = new StringBuilder(2048);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Assign this magic number 2048 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS109)